### ISSUE TITLE
Minecraft Bedrock xbox live friend joining not working on macOS

### ISSUE THREAD

**Issue by vortkk** [minecraft-linux/mcpelauncher-manifest#860]
**Describe the bug**
When I launch the game, go to the friends tab and try to join, the game crashes or the game keeps trying to load forever.

**To Reproduce**
1. Go to the Friends Tab
2. Click on a world from a Friend (Xbox Live)
3. Wait on the "loading world" screen
4. Game Automatically Crashes or it never loads, doesnt join the game

**Expected behavior**
I should be able to join my friends without the game crashing. 

**Screenshots**
<img width="642" alt="Screenshot 2023-07-10 at 2 37 03 AM" src="https://github.com/minecraft-linux/mcpelauncher-manifest/assets/139096193/850e8892-4284-46f9-bcf0-ebdb48f657ce">

**Desktop: 16 Inch, 2019 MacBook Pro**
 - OS: macOS Ventura 13.4.1
 - Version: v0.10.0-macOS-x86_64-0.0.435 (build 435)
 - Version: v0.10.0-macOS-x86_64-0.2.435 (build 435)
 - Installation Type: dmg

**Additional context**
Creating my own world and joining it, isn't a problem. I tried downloading 2 different version and both don't work.

**Comment by crdhs** [comment]
is there any current fix?

**Comment by vortkk** [comment]
> 

Nope

**Comment by vortkk** [comment]
> is there any current fix?

Nope

**Comment by JJRamos2001** [comment]
i am having the same problem i wished it could get worked on sooner

**Comment by vinniedaboi** [comment]
Having the same problem here

**Comment by SNUG-B** [comment]
I am also having this issue. :(

**Comment by Stxvxn-1** [comment]
Same issue here. Every time I try joining a friends server it just crashes my game.

**Comment by vortkk** [comment]
Update: It’s still not working.

**Comment by vortkk** [comment]
Update: It’s still not working with the latest version released last week.

**Comment by ChristopherHX** [comment]
> Update: It’s still not working with the latest version released last week.

Yeah that was a Google Play login hotfix, nothing else I acknowlege the existense of this  bug.

Connecting to a custom bedrock dedicated server works.(the last time I tested this)

> Minecraft Bedrock not working on Mac

This title means for me
- the game does not open
- you cannot play single player
- you cannot play multiplayer

None of them is actually true, because only a single unstable multiplayer method does currently not work on macOS.

**Comment by nikunjsakhrelia** [comment]
Having similar issue with the latest version, i am seeing error in the gamelog regarding bootstrap.json 
can any one help look into this.

<img width="424" alt="Screenshot 2023-08-28 at 10 12 05" src="https://github.com/minecraft-linux/mcpelauncher-manifest/assets/225527/93f64b4d-8f3b-4583-bb85-03f70c8d215f">

**Comment by ChristopherHX** [comment]
@nikunjsakhrelia Please delete the version in settings->versions and download it again, sounds like game assets corruption. (Why? 1.20.30.21 has been verified as working on macOS 10.14 and if everyone with a mac had this issue you wouldn't be the first one reporting it).

If not please open a new issue, this issue is about "xbox live friend joining not working on macOS"

**Comment by nikunjsakhrelia** [comment]
thanks it worked, still cant join friends world, says timeout or he cant join my world

**Comment by ChristopherHX** [comment]
Copy your world to this software https://www.minecraft.net/en-us/download/server/bedrock and connect both clients to it.

As you are on macOS, you can just use docker desktop (expose the required udp / tcp ports) if you don't have any linux/windows device for hosting.
Then add exernal server by ip.

This is the most stable way for multiplayer, without depending on ms servers to be online and working with third party game clients.

**Comment by dohwellington** [comment]
whenever i tried joining a fellow player, it couldnt let me join. it gave me the 'unable to connect' screen every time i tried to join. is there any way this can be fixed without using unnecessary addons?
